lib/chef/knife/ssh.rb

```diff
@@ -362,7 +368,11 @@ def ssh_command(command, subsession = nil)
           if config[:on_error] && exit_status != 0
             chan.close
           else
-            chan.request_pty
+            if config[:ssh_pty]
+              chan.request_pty do |ch, success|
+                raise Train::Transports::SSHPTYFailed, "Requesting PTY failed" unless success
```

Comment:
this is just using net-ssh so definitely this shouldn't be getting raised.  also chef-client should never be raising Train exceptions.

Reply:
Yes, Thanks @lamont-granquist updating that and removing the option `--ssh-pty` entirely.